blockchain/standalone/pow.go

```diff
@@ -218,3 +218,180 @@ func CheckProofOfWork(powHash *chainhash.Hash, difficultyBits uint32, powLimit *
+	// part, n, and a fractional part, f, such that f is in the desired range of
+	// [0,1).  By exponent rules 2^(n + f) = 2^n * 2^f, so the strategy is to
+	// calculate the result by applying the cubic polynomial approximation to
+	// the fractional part and using the fact that multiplying by 2^n is
+	// equivalent to an arithmetic left or right shift depending on the sign.
+	//
+	// In other words, start by calculating the exponent (x) using 64.16 fixed
+	// point and decompose it into integer (n) and fractional (f) parts as
+	// follows:
+	//
+	//       2^16 * (Δt - Δh*Ib)   (Δt - Δh*Ib) << 16
+	//   x = ------------------- = ------------------
+	//            halfLife              halfLife
+	//
+	//        x
+	//   n = ---- = x >> 16
+	//       2^16
+	//
+	//   f = x (mod 2^16) = x & 0xffff
+	//
+	// The use of 64.16 fixed point for the exponent means both the integer (n)
+	// and fractional (f) parts have an additional factor of 2^16.  Since the
+	// fractional part of the exponent is cubed in the polynomial approximation
+	// and (2^16)^3 = 2^48, the addition step in the approximation is internally
+	// performed used 16.48 fixed point to compensate.
+	//
+	// In other words, the fixed point formulation of the goal cubic polynomial
+	// approximation for the fractional part is:
+	//
+	//                 195766423245049*f + 971821376*f^2 + 5127*f^3 + 2^47
+	//   2^f ~= 2^16 + ---------------------------------------------------
+	//                                          2^48
+	//
+	// Finally, the final target difficulty is calculated using x.16 fixed point
+	// and then clamped to the valid range as follows:
+	//
+	//              startDiff * 2^f * 2^n
+	//   nextDiff = ---------------------
+	//                       2^16
+	//
+	//   nextDiff = min(max(nextDiff, 1), powLimit)
+	//
+	// NOTE: The division by the half life uses Quo instead of Div because it
+	// must be truncated division (which is truncated towards zero as Quo
+	// implements) as opposed to the Euclidean division that Div implements.
+	idealTimeDelta := heightDelta * targetSecsPerBlock
+	exponentBig := big.NewInt(timeDelta - idealTimeDelta)
+	exponentBig.Lsh(exponentBig, 16)
+	exponentBig.Quo(exponentBig, big.NewInt(halfLife))
+
+	// Decompose the exponent into integer and fractional parts.  Since the
+	// exponent is using 64.16 fixed point, the bottom 16 bits are the
+	// fractional part and the integer part is the exponent right shifted by 16.
+	frac64 := uint64(exponentBig.Int64() & 0xffff)
+	shifts := exponentBig.Rsh(exponentBig, 16).Int64()
+
+	// Calculate 2^16 * 2^(fractional part) of the exponent.
+	//
+	// Note that a full unsigned 64-bit type is required to avoid overflow in
+	// the internal 16.48 fixed point calculation.  Also, the overall result is
+	// guaranteed to be positive and a maximum of 17 bits, so it is safe to cast
+	// to a uint32.
+	const (
+		polyCoeff1 uint64 = 195766423245049 // 0.695502049 * 2^48
```

Comment:
Notice that double checking the math in the comment gives `ceil(0.695502049 * 2^48) = 195766423044489`  which is not `195766423245049`.  This is because it was calculated based on a more precise version of the polynomial than the one in the comments.

This comment should be updated to the more precise value of `ceil(0.695502049712533 * 2^48)` to avoid confusion.

Reply:
Updated.